library/src/main/java/com/mapzen/android/graphics/OverlayManager.java

```diff
@@ -161,14 +161,18 @@ public void setMyLocationEnabled(boolean enabled) {
    * Optionally persists the map data to the {@link MapDataManager}.
    */
   public void setMyLocationEnabled(boolean enabled, boolean persistMapData) {
```

Comment:
We should also add a check to see if the value of `enabled` changes so that we avoid adding two `PersistableMapData` objects when `enabled` is `true`. 

```
boolean changed = myLocationEnabled != enabled;
    if (!changed) {
      return;
    }
    myLocationEnabled = enabled;
```